src/Maker/MakeResetPassword.php

```diff
@@ -194,23 +203,84 @@ public function generate(InputInterface $input, ConsoleStyle $io, Generator $gen
             ]
         );
 
-        $generator->generateClass(
-            $requestClassNameDetails->getFullName(),
-            'resetPassword/ResetPasswordRequest.tpl.php',
-            [
-                'repository_class_name' => $repositoryClassNameDetails->getFullName(),
-                'user_full_class_name' => $userClassNameDetails->getFullName(),
-            ]
+        $entityClassGenerator = new EntityClassGenerator($generator, $this->doctrineHelper);
+
+        $pathRequestEntity = $entityClassGenerator->generateEntityClass($requestClassNameDetails, false);
+
+        $generator->writeChanges();
+
+        $manipulator = new ClassSourceManipulator(
+            $this->fileManager->getFileContents($pathRequestEntity)
         );
 
-        $generator->generateClass(
-            $repositoryClassNameDetails->getFullName(),
-            'resetPassword/ResetPasswordRequestRepository.tpl.php',
-            [
-                'request_class_full_name' => $requestClassNameDetails->getFullName(),
-                'request_class_name' => $requestClassNameDetails->getShortName(),
-            ]
+        $manipulator->addInterface(ResetPasswordRequestInterface::class);
+
+        $manipulator->clearClassNodeStmts();
+
+        $manipulator->addTrait(ResetPasswordRequestTrait::class);
+
+        $manipulator->addConstructor([
+                ['user', null, 'object'],
+                ['expiresAt', null, '\DateTimeInterface'],
+                ['selector', null, 'string'],
+                ['hashedToken', null, 'string'],
```

Comment:
Instead of `$params` being an array of arrays, could we use the `PhpParser\Builder\Param` class instead? I think it would look like this:

```php
$manipulator->addConstructor([
    (new Param('user')->setType('object')),
    (new Param('expiresAt')->setType(\DateTimeInterface::class)),
],
// ...
```

Then we would just need to update the `ClassSourceManipulator` to call `->getNode()` on each of these Param "Builder" classes to convert it into a real `Param` node.

Reply:
We can add directly nodes :

Reply:
We can pass directly nodes, this is what i wanted to do in the beginning

Reply:
You're right, we should just pass nodes directly. But, our code, let's *use* the "builder" - it's more convenient, or at least more expressive :) 

```php
$manipulator->addConstructor([
    (new Param('user')->setType('object'))->getNode(),
    (new Param('expiresAt')->setType(\DateTimeInterface::class))->getNode(),
],
```

Reply:
Ha okay, i did'nt know that we can use builder like this. I used instance of builder like that. Great, it is very clear, you right.